### PR TITLE
[HIVE-22098] Data loss occurs when multiple tables are join with different bucketing_version

### PR DESCRIPTION
**Pull request by jithendhir92**

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecMapper.java [hunk 1]
```diff
@@ -20,10 +20,12 @@
 
 import java.io.IOException;
 import java.net.URLClassLoader;
+import java.util.ArrayList;
```
Comment (sudhir-reddy) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecMapper.java]: need to bump up the pom build versions too ?

File: ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecMapper.java [hunk 2]
```diff
@@ -138,6 +143,43 @@ public void configure(JobConf job) {
...
+  /**
+   * defined-self balance ReduceSinkOperator of bucketVersion, keep values to sameness
+   * @param rootOp
+   */
+  private static void balanceRSOpbucketVersion(Operator rootOp){
+    List<Operator<? extends OperatorDesc>> needDealOps = new ArrayList<Operator<? extends OperatorDesc>>();
+    visitChildGetRSOps(rootOp, needDealOps);
+    int bucketVersion = -1;
+    for(Operator<? extends OperatorDesc> rsop : needDealOps){
+      if(rsop.getBucketingVersion() != 2 && rsop.getBucketingVersion() != 1){
```
Comment (sudhir-reddy) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecMapper.java]: isn't it safe to do it as "and not greater than 2" ?
ideally if we change it to:  if it is zero or -1 use -1 ?

Comment (sudhir-reddy) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecMapper.java]: can we also see if we can do this check inside the place where the bucketing algorithm is being picked up ?

File: ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecMapper.java (hunk 1 above)
Comment (jithendhir92) [reply ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecMapper.java]: Ack. Version bump up is required. Will push the changes

**Pull request closed without merging**